### ISSUE TITLE
Inconsistency about refractory period with Brian Simulator

### ISSUE THREAD

**Issue by ssmint** [Hananel-Hazan/bindsnet#137]
In my opinion, implementation of refractory period in Brian Simulation makes more sense.

Assume refrac_period = 1 
```
Bindsnet
t=0: mem+=input --> neuron spikes --> refrac = 1
t=1: refrac = 0 --> mem+=input --> neuron spikes 
```
Ref:
https://github.com/Hananel-Hazan/bindsnet/blob/6d4a7a7980080556c79c34d2a603bada12dc78d0/bindsnet/network/nodes.py#L324-L334

```
Brian simulation
t=0: mem+=input --> neuron spikes --> t_spike = 0
t=1: t_current - t_spike < refrac_period  --> do nothing
t=2: t_current - t_spike > refrac_period --> mem+=input --> neuron spikes --> ...
```
Ref:
https://github.com/brian-team/brian2/blob/2b8e459798bd84be1c01e707d74993f2f260b5ce/brian2/groups/neurongroup.py#L121

**Comment by ssmint** [comment]
Since there is a definition of _t_ in [network/__init__](https://github.com/Hananel-Hazan/bindsnet/blob/6d4a7a7980080556c79c34d2a603bada12dc78d0/bindsnet/network/__init__.py#L246), one possible way is to add _t_ as an additional argument to [step](https://github.com/Hananel-Hazan/bindsnet/blob/6d4a7a7980080556c79c34d2a603bada12dc78d0/bindsnet/network/__init__.py#L252) function. Inside the step function, introduce new variable to store spiking time (say _t_spike_). Everytime step function is called, check if _t_-_t_spike_ is greater than refractory period and do operation accordingly.

**Comment by djsaunde** [comment]
Hm, that could be problematic when doing multiple simulations in sequence. For example:

```
network.run(inpts, time=100)
network.run(inpts, time=100)  # Forgets refractory neurons from end of previous simulation!
```

Alternatively, one could just reverse the ordering, from:

```
# Decrement refractory counters.
self.refrac_count[self.refrac_count != 0] -= dt 

# Integrate inputs. 
self.v += (self.refrac_count == 0).float() * inpts 
  
# Check for spiking neurons.
self.s = self.v >= self.thresh

# Refractoriness and voltage reset. 
self.refrac_count.masked_fill_(self.s, self.refrac) 
```

to

```
# Integrate inputs. 
self.v += (self.refrac_count == 0).float() * inpts 
  
# Decrement refractory counters.
self.refrac_count[self.refrac_count != 0] -= dt 

# Check for spiking neurons.
self.s = self.v >= self.thresh

# Refractoriness and voltage reset. 
self.refrac_count.masked_fill_(self.s, self.refrac) 
```

Does this make sense? Does this appear to solve the problem?

**Comment by ssmint** [comment]
Yes, that will be better than my solution.

**Comment by ssmint** [comment]
You may need to be careful when changing order. There are models (like [this](https://github.com/Hananel-Hazan/bindsnet/blob/6d4a7a7980080556c79c34d2a603bada12dc78d0/bindsnet/network/nodes.py#L564-L574)) that do self.refrac_count == 0 more than once.

**Comment by djsaunde** [comment]
Solved by #136.
